packages/@vuepress/core/src/types/page.ts

```diff
@@ -6,141 +6,141 @@ import type { PageData, PageFrontmatter } from '@vuepress/shared'
-  ExtraPageFields extends Record<any, any> = Record<never, never>
-> = PageData &
-  ExtraPageFields & {
-    /**
-     * Data of the page, which will be available in client code
-     */
-    data: PageData<ExtraPageData>
-
-    /**
-     * Raw Content of the page
-     */
-    content: string
-
-    /**
-     * Rendered content of the page
-     */
-    contentRendered: string
-
-    /**
-     * Date of the page, in 'yyyy-MM-dd' format
-     *
-     * @example '2020-09-09'
-     */
-    date: string
-
-    /**
-     * Dependencies of the page
-     */
-    deps: string[]
-
-    /**
-     * Hoisted tags of the page
-     */
-    hoistedTags: string[]
-
-    /**
-     * Links of the page
-     */
-    links: MarkdownLink[]
-
-    /**
-     * Path of the page that inferred from file path
-     *
-     * If the page does not come from a file, it would be `null`
-     *
-     * @example '/guide/index.html'
-     */
-    pathInferred: string | null
-
-    /**
-     * Locale path prefix of the page
-     *
-     * @example '/getting-started.html' -> '/'
-     * @example '/en/getting-started.html' -> '/en/'
-     * @example '/zh/getting-started.html' -> '/zh/'
-     */
-    pathLocale: string
-
-    /**
-     * Permalink of the page
-     *
-     * If the page does not have a permalink, it would be `null`
-     */
-    permalink: string | null
-
-    /**
-     * Custom data to be attached to the page route record of vue-router
-     *
-     * @see https://next.router.vuejs.org/api/#meta
-     */
-    routeMeta: Record<string, unknown>
-
-    /**
-     * Slug of the page
-     */
-    slug: string
-
-    /**
-     * Source file path
-     *
-     * If the page does not come from a file, it would be `null`
-     */
-    filePath: string | null
-
-    /**
-     * Source file path relative to source directory
-     *
-     * If the page does not come from a file, it would be `null`
-     */
-    filePathRelative: string | null
-
-    /**
-     * Component file path
-     */
-    componentFilePath: string
-
-    /**
-     * Component file path relative to temp directory
-     */
-    componentFilePathRelative: string
-
-    /**
-     * Component file chunk name
-     *
-     * Only take effect in webpack
-     */
-    componentFileChunkName: string
-
-    /**
-     * Page data file path
-     */
-    dataFilePath: string
-
-    /**
-     * Page data file path relative to temp directory
-     */
-    dataFilePathRelative: string
-
-    /**
-     * Page data file chunk name
-     *
-     * Only take effect in webpack
-     */
-    dataFileChunkName: string
-
-    /**
-     * Rendered html file path
-     */
-    htmlFilePath: string
-
-    /**
-     * Rendered html file path relative to dest directory
-     */
-    htmlFilePathRelative: string
-  }
+  ExtraPageFields extends Record<any, any> = Record<never, never>,
+  Frontmatter extends PageFrontmatter = PageFrontmatter
+> = PageData<ExtraPageFields, Frontmatter> & {
```

Comment:
`ExtraPageFields` may not be part of `PageData`.

`page.foo` and `page.data.foo` are different

Reply:
I understand your idea, but I am not seeing any difference between this change. The "Page Data is actually `page.data` on behind lines.

If you really want to declare a difference here, maybe you should copy all the fields in `PageData` and place here the second time. If you just extend it, then I would just regard it as an interface with no special meanings (Just a shape of some basic page properties), so that's why I place `ExtraPageFields` on it.

Reply:
Or in another word, we rename current `PageData` as `BasicPage`

And we decalre `type PageData<...> = BasicPage<...>`

Then we wrote

```
type Page<...> = BasicPage & ExtraPageFields & {
  ...
  data: PageData<...>
}
```

Reply:
I update the types to try making a better declaration.